developing-without-a-build/article.md

```diff
@@ -0,0 +1,257 @@
+
+These tools can be really complex, projects can set up systems which transform different syntaxes and dialects into bundles we can run in the browser. Source maps need to be configured carefully between the different tools, so that we can still recognize and debug our code in the browser.
+
+Imagine a new person coming into web development, even experienced developers, they first need to learn about a lot of different tools before they can get to actual development. Their IDE and linter don't understand the syntax of this framework that was recommended to them by a friend. Installing a few extra plugins solved that. After that they ran into some more errors... Turns out they wanted to keep it simple and not use typescript, but they didn't configure the babel plugins for decorators correctly in the right order so the project didn't compile. Oops!
+
+It sounds exaggerated, but these things are really happening to people. I feel that's really a shame. One of the key selling points of the web is that's an easy and open format. It should also be easy to just get started right away without a lot of ceremony.
+
+I'm not criticizing any of the tools, they have a role and a purpose. And for a long time it was the only real way to build complex applications on the web, as the web standards and browser implementations were just not there to support modern web development. They really helped push web development forward.
+
+However nowadays I think build tools and bundlers should be primarily for production optimization, for development I think we can go back to basics. Browser support for es module features has improved a lot in recent years.
+
+In this article we will explore different ways of loading modules in the browser, and offer practical advice on how to handle different scenarios. We will start by looking at javascript modules, and later look at html, css and json as well.
+
+## Loading modules in the browser
+This is not exact step by step tutorial, but you can follow along with any of the examples by using any simple web server. For example `http-server` from npm. Run it with `-c-1` to disable time based caching.
+
+```bash
+npm i -D http-server
+npx http-server -o -c-1
+```
+
+### Loading modules
+Modules are loaded in the browser using regular script tags with a `type="module"` attribute. We can just write our module code directly inline:
+```html
+<!DOCTYPE html>
+<html>
+<head></head>
+
+<body>
+  <script type="module">
+    console.log('hello world!');
+  </script>
+</body>
+
+</html>
+```
+
+From here we can use static imports to load other modules:
+```html
+<script type="module">
+  import './app.js';
+
+  console.log('hello world!');
+</script>
+```
+
+Note that we need to use an explicit file extension, as the browser doesn't know which file to request otherwise.
```

Comment:
```suggestion
Note that we need to use an explicit file extension, as the browser doesn't know which file to request otherwise, so if you're used to writing imports like `import _ from 'lodash-es'`, you'll have to change them to `import _ from '/node_modules/lodash-es/index.js''`. More on that later.
```